.github/workflows/branch.yml

```diff
@@ -3,63 +3,71 @@ name: Branch
 on:
   push:
     branches-ignore:
-      - 'master'
+      - master
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
 jobs:
 
-  run_tests:
-    name: Test (${{ matrix.python-version }}, ${{ matrix.os }})
+  support-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: set-matrix
+        run: echo "::set-output name=matrix::$(python matrix.py)"
```

Comment:
I may not be fully understanding this, but I think there is a disadvantage to having to look at a different file to figure out what OS's, etc. are being tested. I'm thinking that something like [this](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) should work for us:
```
runs-on: ${{ matrix.os }}
strategy:
  matrix:
    os: [macos-latest, windows-latest, ubuntu-latest]
    python: [3.7, 3.8, 3.9, 4.0]
    ....
````

Reply:
Yeah, it's not great but I wanted to be able to specify the same matrix in multiple separate workflows without hard-coding the values. I couldn't find a way to just have a composite action specify it since the matrix is a top level job parameter and needs to be specified outside of individual steps within a job in the workflow. Happy to work through suggestions and was going to be more explicit e.g. `cicd-matrix.py` at least.